libs/eavmlib/src/esp.erl

```diff
@@ -177,9 +177,9 @@ nvs_get_binary(Key) when is_atom(Key) ->
 %%-----------------------------------------------------------------------------
 -spec nvs_get_binary(Namespace :: atom(), Key :: atom()) -> binary() | undefined.
 nvs_get_binary(Namespace, Key) when is_atom(Namespace) andalso is_atom(Key) ->
-    case nvs_fetch_binary(Namespace, Key) of
+    case esp:nvs_fetch_binary(Namespace, Key) of
```

Comment:
Doesn’t seem necessary, but ok.

Reply:
It is, we need to get the function resolved as an `nvs_fetch_binary` external function, otherwise the local version is called that will run always nif_error.